commit cb90e3d9556fd41bc7f5ec06c048f2ca752669dc
Author: Umaro <umaro@narshe.net>
Date:   Tue Jun 21 21:49:55 2011 +0200

    mm/dmapool.c: use TASK_UNINTERRUPTIBLE in dma_pool_alloc()
    
        As it stands this code will degenerate into a busy-wait if the calling task
        has signal_pending().
    
        Cc: Rolf Eike Beer <eike-kernel@sf-tec.de>
        Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
        Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

diff --git a/mm/dmapool.c b/mm/dmapool.c
index e9446d9..55bbfa9 100644
--- a/mm/dmapool.c
+++ b/mm/dmapool.c
@@ -322,7 +322,7 @@ void *dma_pool_alloc(struct dma_pool *pool, gfp_t mem_flags,
 		if (mem_flags & __GFP_WAIT) {
 			DECLARE_WAITQUEUE(wait, current);
 
-			__set_current_state(TASK_INTERRUPTIBLE);
+			__set_current_state(TASK_UNINTERRUPTIBLE);
 			__add_wait_queue(&pool->waitq, &wait);
 			spin_unlock_irqrestore(&pool->lock, flags);
 
